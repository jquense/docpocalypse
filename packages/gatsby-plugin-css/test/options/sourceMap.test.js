const path = require('path');

const { webpack } = require('../helpers');

describe('Options', () => {
  test('Sourcemap - {Boolean}', () => {
    const config = {
      devtool: 'sourcemap',
      loader: {
        test: /\.css$/,
        options: {
          sourceMap: true,
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual(
        'module.exports = "a { color: rgba(255, 0, 0, 1.0) }\\n"',
      );

      expect(source).toMatchSnapshot();

      // eslint-disable-next-line no-underscore-dangle
      const map = stats.compilation.modules[1]._source._sourceMap;

      map.file = path.posix.relative(__dirname, map.file);
      map.sources = map.sources.map((src) =>
        path.posix.relative(__dirname, src),
      );

      expect(map).toMatchSnapshot();
    });
  });

  test('Sourcemap - {String}', () => {
    const config = {
      loader: {
        test: /\.css$/,
        options: {
          sourceMap: 'inline',
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual(
        'module.exports = "a { color: rgba(255, 0, 0, 1.0) }\\n\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInRlc3QvZml4dHVyZXMvY3NzL3N0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxJQUFJLDRCQUFhIiwiZmlsZSI6InRlc3QvZml4dHVyZXMvY3NzL3N0eWxlLmNzcyIsInNvdXJjZXNDb250ZW50IjpbImEgeyBjb2xvcjogYmxhY2sgfVxuIl19 */"',
      );

      expect(source).toMatchSnapshot();
    });
  });
});
