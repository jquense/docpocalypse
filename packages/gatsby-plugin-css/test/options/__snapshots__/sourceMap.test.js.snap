// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Sourcemap - {Boolean} 1`] = `"module.exports = \\"a { color: rgba(255, 0, 0, 1.0) }\\\\n\\""`;

exports[`Options Sourcemap - {Boolean} 2`] = `
Object {
  "file": "../fixtures/css/style.css",
  "mappings": "AAAA,IAAI,4BAAa",
  "names": Array [],
  "sources": Array [
    "../fixtures/css/style.css",
  ],
  "sourcesContent": Array [
    "a { color: black }
",
  ],
  "version": 3,
}
`;

exports[`Options Sourcemap - {String} 1`] = `"module.exports = \\"a { color: rgba(255, 0, 0, 1.0) }\\\\n\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInRlc3QvZml4dHVyZXMvY3NzL3N0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxJQUFJLDRCQUFhIiwiZmlsZSI6InRlc3QvZml4dHVyZXMvY3NzL3N0eWxlLmNzcyIsInNvdXJjZXNDb250ZW50IjpbImEgeyBjb2xvcjogYmxhY2sgfVxuIl19 */\\""`;
